gui/WorldView.cxx

```diff
@@ -976,6 +978,59 @@ void WorldView::computeContour(double threshold)
   }
 }
 
+//-----------------------------------------------------------------------------
+void WorldView::loadMesh(QString const& path)
+{
+  QFileInfo check_file(path);
+  if (!check_file.exists() || !check_file.isFile())
+  {
+    return;
+  }
+  // Create the vtk pipeline
+  // Read volume
+  vtkNew<vtkPLYReader> reader;
+  reader->SetFileName(qPrintable(path));
+  reader->Update();
+
+  vtkSmartPointer<vtkPolyData> mesh = reader->GetOutput();
+  this->setMesh(mesh);
+}
+
+//-----------------------------------------------------------------------------
+void WorldView::setMesh(vtkSmartPointer<vtkPolyData> mesh)
+{
+  QTE_D();
+
+  mesh->GetPointData()->SetActiveScalars("RGB");
+
+  // Create mapper
+  vtkNew<vtkPolyDataMapper> meshMapper;
+  meshMapper->SetInputData(mesh);
+  meshMapper->ScalarVisibilityOn();
+  meshMapper->SetScalarModeToUsePointFieldData();
+  meshMapper->SelectColorArray("RGB");
+  meshMapper->Update();
+
+  // Set the actor's mapper
+  d->volumeActor->SetMapper(meshMapper);
+  d->volumeActor->SetVisibility(true);
+  d->volumeOptions->setActor(d->volumeActor);
+  d->volumeOptions->setEnabled(true);
+
+  if (d->UI.actionShowVolume->isChecked())
+  {
+    this->fusedMeshEnabled(true);
+  }
+  connect(d->UI.actionShowVolume, &QAction::triggered,
+          this, &WorldView::fusedMeshEnabled);
```

Comment:
I don't know much about this stuff.  I just copied this from `setVolume`, which does the same thing.  I think the intent was that Volume options are not enabled until we set data that they can operate on them.

It's not quite right to use volume options here anyway because while the volume options are mostly mesh options there is at least one setting, the surface threshold, that requires a volume.  A better solution would be to have separate toolbar buttons for volume and mesh (instead of mesh options nested under the volume), but I don't have the time or expertise to do that.  Also, there is currently no visualization of the volume without the mesh, but there could be.  I often use paraview to do volume rendering of the volume data.   It would be nice to have that in TeleSculptor.  These changes are all outside the scope of this PR.

Reply:
Yeah, I saw; I don't think `setVolume` is doing it right either. Some other issues:
- If you set a volume and then a mesh, or just set either more than once, you'll wind up with multiple connections and possibly emitting the signal multiple times. (I'm actually not sure if Qt de-duplicates connections...) Although that's already true due to connecting to both `triggered` and `toggled`.
- If the mesh/volume is unloaded (not sure this is currently supported?), the signal will still be emitted.

Agreed that splitting mesh and volume is probably TRTTD but could be punted for now.

Reply:
We do have a `resetVolme()` that includes
```
this->fusedMeshEnabled(false);
d->volumeOptions->setEnabled(false);
```

I don't remember why the `connect` calls are here or if there is a good reason for that.  I also don't remember why we have both `toggled` and `triggered`.  What is the difference?

I'm not the right person to be making Qt changes at this level of Qt depth.  Maybe I could ask you to just fix the Qt aspects of this on a new branch for the `setVolume` case and I'll mirror that approach here?  Or do this for both after we land this branch?

Reply:
> I don't remember why the `connect` calls are here

Again, my guess is so that the signal isn't emitted until a volume (or, now, mesh) is actually loaded.

>  I also don't remember why we have both `toggled` and `triggered`. What is the difference?

Well, [the documentation](https://doc.qt.io/qt-5/qaction.html) would answer that :slightly_smiling_face:. Short version, though: `setChecked` won't fire `triggered`. AFAICT, though, for a checkable action, anything that will fire `triggered` will also fire `toggled`, which is why I'm fairly sure connecting to both is pointless.

Reply:
Eek... okay, took a look at doing that, and it's even worse than I realized; toggling the visibility via `setVolumeVisible` fails to emit the signal entirely. PR coming soon.